packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -37,174 +45,238 @@ export const GridRootStyles = styled('div', {
-    { [`& .${gridClasses.withBorderColor}`]: styles.withBorderColor },
-    { [`& .${gridClasses.treeDataGroupingCell}`]: styles.treeDataGroupingCell },
-    { [`& .${gridClasses.treeDataGroupingCellToggle}`]: styles.treeDataGroupingCellToggle },
-    { [`& .${gridClasses.detailPanelToggleCell}`]: styles.detailPanelToggleCell },
+      [`& .${c['columnHeader--withRightBorder']}`]: styles['columnHeader--withRightBorder'],
+    },
+    { [`& .${c.columnHeader}`]: styles.columnHeader },
+    { [`& .${c.headerFilterRow}`]: styles.headerFilterRow },
+    { [`& .${c.columnHeaderCheckbox}`]: styles.columnHeaderCheckbox },
+    { [`& .${c.columnHeaderDraggableContainer}`]: styles.columnHeaderDraggableContainer },
+    { [`& .${c.columnHeaderTitleContainer}`]: styles.columnHeaderTitleContainer },
+    { [`& .${c['columnSeparator--resizable']}`]: styles['columnSeparator--resizable'] },
+    { [`& .${c['columnSeparator--resizing']}`]: styles['columnSeparator--resizing'] },
+    { [`& .${c.columnSeparator}`]: styles.columnSeparator },
+    { [`& .${c.filterIcon}`]: styles.filterIcon },
+    { [`& .${c.iconSeparator}`]: styles.iconSeparator },
+    { [`& .${c.menuIcon}`]: styles.menuIcon },
+    { [`& .${c.menuIconButton}`]: styles.menuIconButton },
+    { [`& .${c.menuOpen}`]: styles.menuOpen },
+    { [`& .${c.menuList}`]: styles.menuList },
+    { [`& .${c['row--editable']}`]: styles['row--editable'] },
+    { [`& .${c['row--editing']}`]: styles['row--editing'] },
+    { [`& .${c['row--dragging']}`]: styles['row--dragging'] },
+    { [`& .${c.row}`]: styles.row },
+    { [`& .${c.rowReorderCellPlaceholder}`]: styles.rowReorderCellPlaceholder },
+    { [`& .${c.rowReorderCell}`]: styles.rowReorderCell },
+    { [`& .${c['rowReorderCell--draggable']}`]: styles['rowReorderCell--draggable'] },
+    { [`& .${c.sortIcon}`]: styles.sortIcon },
+    { [`& .${c.withBorderColor}`]: styles.withBorderColor },
+    { [`& .${c.treeDataGroupingCell}`]: styles.treeDataGroupingCell },
+    { [`& .${c.treeDataGroupingCellToggle}`]: styles.treeDataGroupingCellToggle },
+    { [`& .${c.detailPanelToggleCell}`]: styles.detailPanelToggleCell },
     {
-      [`& .${gridClasses['detailPanelToggleCell--expanded']}`]:
-        styles['detailPanelToggleCell--expanded'],
+      [`& .${c['detailPanelToggleCell--expanded']}`]: styles['detailPanelToggleCell--expanded'],
     },
     styles.root,
   ],
-})<{ ownerState: OwnerState }>(({ theme }) => {
-  const borderColor = getBorderColor(theme);
-  const radius = theme.shape.borderRadius;
+})<{ ownerState: OwnerState }>(({ theme: t }) => {
+  const borderColor = getBorderColor(t);
+  const radius = t.shape.borderRadius;
+
+  const containerBackground = t.vars
+    ? t.vars.palette.background.defaultChannel
+    : t.palette.background.default;
+
+  const pinnedBackground =
+    t.palette.mode === 'dark'
+      ? lighten(
+          t.vars ? t.vars.palette.background.defaultChannel : t.palette.background.default,
```

Comment:
JS color manipulations won't work with `t.vars` as `t.vars.palette.background.defaultChannel` contains a CSS variable.
This is the reason https://deploy-preview-10059--material-ui-x.netlify.app/x/react-data-grid/experiments/joy-ui/ throws an error

Reply:
I don't understand theme much yet, would you have a proposition to solve this issue? Should I pick a different field from the theme object?

Reply:
I'll take a look

Reply:
I've pushed a commit with a fix, but there's no support for darken/lighten when using CSS variables.
So when the data grid is used inside `CSSVarProvider` (instead of regular `ThemeProvider`), `pinnedBackground === containerBackground`.
I think it's fine since we might remove different background color for pinned columns/rows anyway as discussed in https://github.com/mui/mui-x/pull/10059#discussion_r1417215551

Reply:
It's not too bad though - most of the changes are expected.
I've added comments in the PR for those that seem to be out of place

Reply:
One thing with your change though, the hover & selected styles aren't blended with the background color, it's just the color raw. Is there a way we can fix those? We haven't really discussed, but what would be the recommended way to customize that sort of color for the grid?

Reply:
> One thing with your change though, the hover & selected styles aren't blended with the background color, it's just the color raw. Is there a way we can fix those?

Well, I won't be able to recreate the same blend, but I can use [`color-mix`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-mix). Browser support isn't great though (version-wise).

> We haven't really discussed, but what would be the recommended way to customize that sort of color for the grid?

What is the current approach? Users have to override colors for hover, selected, and selected on hover?

Reply:
The current approach would be to use the MUI themes. I think it's ok to leave it as it is and adjust later though.